cpp/pipelines/add_authority_beacon_information.cc

```diff
@@ -69,10 +128,23 @@ int Main(int argc, char **argv) {
     auto authority_writer(MARC::Writer::Factory(augmented_authority_records_filename));
 
     std::vector<BeaconFile> beacon_files;
-    for (int arg_no(3); arg_no < argc; ++arg_no)
-        beacon_files.emplace_back(BeaconFile(argv[arg_no]));
+    std::map<std::string, TypeFile> beacon_to_type_files_map;
+
+
+    for (int arg_no(3); arg_no < argc; ++arg_no) {
+        if (argv[arg_no] == std::string("--type-file")) {
```

Comment:
You can just use strcmp instead of creating two temporary std::string instances.

Reply:
Done